### DISCUSSION TITLE
docs: populate Collection from entity constructor

### DISCUSSION THREAD

**Discussion by mjroeleveld** [mikro-orm/mikro-orm#4300] [Q&A]
I can't find this anywhere unfortunately. Following the example from the docs, how to populate set `tags` from within the constructor? It seems adding `tags: Collection<BookTag>` to the parameter list doesn't cut it; it stays undefined. I am using `forceEntityConstructor: true`.

```typescript
@Entity()
export class Book {

  @PrimaryKey()
  _id: ObjectID;

  @Property()
  title: string;

  @ManyToOne()
  author: Author;

  @ManyToOne()
  publisher: Publisher;

  @ManyToMany({ entity: () => BookTag, inversedBy: 'books' })
  tags = new Collection<BookTag>(this);

  constructor(title: string, author: Author) {
    this.title = title;
    this.author = author;
  }

}
```

**Comment by B4nan** [comment]
I don't understand what you mean, you can't "populate a collection from constructor", that is an async action, and constructors are synchronous.

**Reply by B4nan** [reply]
Maybe you want eager loading? Although I would discourage that on collections, it can cause performance issues if your collections get large over time.

```ts
@ManyToMany({ entity: () => BookTag, inversedBy: 'books', eager: true })
tags = new Collection<BookTag>(this);
```

**Reply by mjroeleveld** [reply]
I mean, a `findOne` with `populate: ['tags']` does not result in `tags` being passed into the constructor and thus there seems to be no way to set `tags`.

**Reply by B4nan** [reply]
Well, it shouldn't, the question is why do you expect that? The entity gets constructed without relations, those are populated later.

**Reply by mjroeleveld** [reply]
Okay, I am noticing that my relations are not populated so I expected the issue to be related to this. But I guess it will be in the decoration then. My situation is exactly the same is the example though, which is why I am confused.

**Reply by B4nan** [reply]
So please provide a reproduction, entity definition is not enough, it needs to be complete so I can run it myself. 

Btw does it work _without_ the `forceEntityConstructor: true`? I guess it should be irrelevant, as that flag is about creating new entity instances - population operates on already created entities, those two things are separate tasks.
